hal/source/bmp180_device.cpp

```diff
@@ -0,0 +1,336 @@
+	if (BusRead(kPROM_START_ADDR, kPROM_DATA_LEN, params_buffer) != kNoError)
+		return make_error_condition(errc::io_error);
+
+	calib_param_.ac1 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC1_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC1_LSB]);
+	calib_param_.ac2 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC2_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC2_LSB]);
+	calib_param_.ac3 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_AC3_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC3_LSB]);
+	calib_param_.ac4 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC4_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC4_LSB]);
+	calib_param_.ac5 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC5_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_AC5_LSB]);
+	calib_param_.ac6 = static_cast<unsigned int>(\
+			params_buffer[kCALIB_PARAM_AC6_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS| params_buffer[kCALIB_PARAM_AC6_LSB]);
+	calib_param_.b1 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_B1_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_B1_LSB]);
+	calib_param_.b2 = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_B2_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_B2_LSB]);
+	calib_param_.mb = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MB_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MB_LSB]);
+	calib_param_.mc = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MC_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MC_LSB]);
+	calib_param_.md = static_cast<int>(\
+			params_buffer[kCALIB_PARAM_MD_MSB] << \
+			kSHIFT_BIT_POSITION_BY_08_BITS | params_buffer[kCALIB_PARAM_MD_LSB]);
+	return kNoError;
+}
+
+error_condition Bmp180Device::GetTemperature(int& temperature) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	int  uncomp_temperature, x1, x2 = 0;
+
+	if (GetUncompTemperature(uncomp_temperature) != kNoError) {
+		return make_error_condition(errc::io_error);
+	}
+	// calculate temperature
+	x1 = ((uncomp_temperature - static_cast<int>(calib_param_.ac6)) * \
+			static_cast<int>(calib_param_.ac5)) >> \
+					kSHIFT_BIT_POSITION_BY_15_BITS;
+	if (x1 == kCHECK_DIVISOR && calib_param_.md == kCHECK_DIVISOR)
+		return kInvalidArgument;
+
+	// executed only the divisor is not zero
+	x2 = (static_cast<int>(calib_param_.mc) << kSHIFT_BIT_POSITION_BY_11_BITS) /\
+			(x1 + calib_param_.md);
+	param_b5_ = x1 + x2;
+	temperature = ((param_b5_ + kCALCULATE_TRUE_TEMPERATURE) >> \
+			kSHIFT_BIT_POSITION_BY_04_BITS);
+
+	return kNoError;
+}
+
+error_condition Bmp180Device::GetPressure(int& pressure) {
+	if (!IsInitialized())
+		return make_error_condition(errc::no_such_device);
+
+	int uncomp_pressure, x1, x2, x3, b3, b6 = 0;
```

Comment:
Please have some comments on these variables, what are they used for, what's their purpose.